src/charm.py

```diff
@@ -217,11 +217,9 @@ def _on_config_changed(self, event: EventBase):
             return
         if not self._bessd_container.can_connect():
             self.unit.status = WaitingStatus("Waiting for bessd container to be ready")
-            event.defer()
```

Comment:
Unfortunately, I do not think this will work because only `ConfigChanged` points to this handler, both `PebbleReady` point to their own handlers. This will need more refactoring.

Reply:
I think we could complety remove both checks for `can_connect()` in this handler.

Reply:
Yes, I agree, I was thinking of handling both pebble ready events using this handler, but we would be in the same situation again of running the configuration script multiple times in a row quickly (For example `bessd_pebble_ready` then `pfcp_pebble_ready`)
However I found that we are checking if the config file has changed or not in and based on that we rewrite the config fiel `upf.json` or not, and we also restart the container or not based on that condition.
Wouldn't it be a better solution if we keep everything as is and we only call `_run_bess_configuration` which will run the `up4` script on the same condition as the restart for example, that being if the config file actually changed we run it, otherwise we don't.
so `self._run_bess_configuration()` would be inside the `if restart:`.